src/user-event/utils/asyncWrapper.ts

```diff
@@ -0,0 +1,35 @@
+import { act } from 'react-test-renderer';
+import { getIsReactActEnvironment, setReactActEnvironment } from '../../act';
+import { flushMicroTasksLegacy } from '../../flush-micro-tasks';
+import { checkReactVersionAtLeast } from '../../react-versions';
+
+export const asyncWrapper = async <T>(
+  callback: () => Promise<T>
+): Promise<T> => {
+  if (checkReactVersionAtLeast(18, 0)) {
+    const previousActEnvironment = getIsReactActEnvironment();
+    setReactActEnvironment(false);
+
+    try {
+      const result = await callback();
+      // Flush the microtask queue before restoring the `act` environment
+      await flushMicroTasksLegacy();
```

Comment:
RTL has addtionally jest advance timers call: https://github.com/testing-library/react-testing-library/blob/c04b8f006c5a683d05c460c8ee1e2248d6f74350/src/pure.js#L49

Do you think we should add this as well?

Reply:
Not sure at all about that, I have no clue what it does. My thought process here was that we already had pretty much our own version of `asyncWrapper` that we use for `waitFor` and it seems to be working pretty well so I figured we might as well stick to that version. Maybe it could be worth investigating what difference it could make to get closer to RTL's implementation but I'd rather do that in another PR